src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let arrayToString = '';
+    const cloneElements = this.concat();
+    const separatorToString = separator + '';
+
+    for (let i = 1; i < cloneElements.length; i += 2) {
+      cloneElements.splice(i, 0, separatorToString);
+    }
```

Comment:
this task you can solved without methods 
for (let i = 0; i < this.length; i++) {
        arrayToString += this[i] + `${separator}`;